gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/modules/LiferayAbstractGradleModuleOperation.java

```diff
@@ -0,0 +1,209 @@
+                    dependencyFileContent.lastIndexOf( "}" ) + 1, dependencyFileContent.length() );
+
+                if( newDenpendencyContent != null && projectDenpendencyContent != null )
+                {
+                    String[] newDenpendencies = newDenpendencyContent.split( "\n" );
+                    StringBuilder preNewContent = new StringBuilder();
+
+                    preNewContent.append( dependencyHead );
+                    preNewContent.append( "dependencies {" );
+                    preNewContent.append( projectDenpendencyContent );
+
+                    for( String newDenpendency : newDenpendencies )
+                    {
+                        if( !newDenpendency.isEmpty() && !projectDenpendencyContent.contains( newDenpendency ) )
+                        {
+                            preNewContent.append( newDenpendency );
+                        }
+                    }
+                    preNewContent.append( "}" );
+                    preNewContent.append( dependencyEnd );
+
+                    if( !dependencyFileContent.equals( preNewContent ) )
+                    {
+
+                        String newContent = preNewContent.toString();
+                        FileUtil.writeFileFromStream(
+                            projectDenpendencyFile, new ByteArrayInputStream( newContent.getBytes() ) );
+
+                    }
+                }
+            }
+        }
+        catch( Exception e )
+        {
+            throw new CoreException( ProjectCore.createErrorStatus( e ) );
+        }
+    }
+    
+    protected String readDenpendencies( File file )
+    {
+        String content = new String( FileUtil.readContents( file, true ) );
+        return content.substring( content.lastIndexOf( "{" ) + 2, content.lastIndexOf( "}" ) );
+    }
+    
+    protected void extractTemplates()
+    {
+        if( !templateFolder.toFile().exists() )
+        {
+            try
+            {
+                final File templateClassFolder = templateFolder.toFile();
+                IPath bladeJarPath = BladeCLI.getBladeCLIPath();
+
+                try(ZipFile zip = new ZipFile( bladeJarPath.toFile() ))
+                {
+                    File temp = File.createTempFile( "templates", ".zip" );
+                    FileUtil.writeFileFromStream( temp, zip.getInputStream( zip.getEntry( "templates.zip" ) ) );
+
+                    if( !templateClassFolder.exists() )
+                    {
+                        ZipUtil.unzip( temp, templateClassFolder );
+                    }
+                }
+                catch( IOException e )
+                {
+                    ProjectCore.logError( e );
+                }
+            }
+            catch( BladeCLIException e )
+            {
+                ProjectCore.logError( e );
+            }
+        }
+    }
+
+    protected File[] getTempateFiles( final String templateName, final String fileType )
```

Comment:
Don't pull templates for these new DS Components from the project templates.  We need to keep the component templates separate since there will not be a 1-to-1 mapping.